lib/deploy.js

```diff
@@ -32,7 +32,20 @@ module.exports = promisify(function(config, callback) {
   var ghPagesConfig = {};
   var rootDir = config.rootDir ? config.rootDir : '.';
   var cloneDir = config.cloneDir ? config.cloneDir : null;
-  var commitMessage = config.message ? config.message : childProcess.execSync('git log --format=%B -n 1').toString().trim();
+
+  var commitMessage = config.message || (function() {
+    var spawn = childProcess.spawnSync('git', ['log', '--format=%B', '-n', '1']);
+    var errorText = spawn.stderr.toString().trim();
+
+    if (errorText) {
+      gutil.log('Fatal error from `git log`.  You must have one commit before deploying.');
+      throw new Error(errorText);
+      callback(spawn.stderr);
```

Comment:
*promisify* should do the right thing both if you throw an error and if you call the callback with the error argument (which is the first argument), so it shouldn't really matter.
Maybe for consistency with the rest of the function, you could call the callback. Note, however, that you're currently calling it after throw, which means it'll never be called. You could instead do:
```JavaScript
if (errorText) {
  gutil.log('Fatal error from `git log`.  You must have one commit before deploying.');
  callback(spawn.stderr);
  return;
}
```